tests/e2e/Services/Projects/ProjectsDevKeys.php

```diff
@@ -0,0 +1,265 @@
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()), []);
+
+        $this->assertEquals(200, $response['headers']['status-code']);
+        $this->assertNotEmpty($response['body']['$id']);
+        $this->assertEquals($keyId, $response['body']['$id']);
+        $this->assertEquals('Key Test', $response['body']['name']);
+        $this->assertNotEmpty($response['body']['secret']);
+        $this->assertArrayHasKey('sdks', $response['body']);
+        $this->assertEmpty($response['body']['sdks']);
+        $this->assertArrayHasKey('accessedAt', $response['body']);
+        $this->assertEmpty($response['body']['accessedAt']);
+
+        /**
+         * Test for FAILURE
+         */
+        $response = $this->client->call(Client::METHOD_GET, '/projects/' . $id . '/development-keys/error', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()), []);
+
+        $this->assertEquals(404, $response['headers']['status-code']);
+
+        return $data;
+    }
+
+    /**
+     * @depends testCreateProject
+     * @group devKeys
+     */
+    public function testNoRateLimitWithDevKey($data): void
+    {
+        $id = $data['projectId'] ?? '';
+
+        /**
+         * Test for SUCCESS
+         */
+        $response = $this->client->call(Client::METHOD_POST, '/projects/' . $id . '/development-keys', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()), [
+            'name' => 'Key Test',
+            'expire' => DateTime::addSeconds(new \DateTime(), 3600),
+        ]);
+
+        $devKey = $response['body']['secret'];
+
+        //
+        for ($i = 0; $i < 11; $i++) {
+            $res = $this->client->call(Client::METHOD_POST, '/account/sessions/email', [
+                'content-type' => 'application/json',
+                'x-appwrite-project' => $id,
+            ], [
+                'email' => 'user@appwrite.io',
+                'password' => 'password'
+            ]);
+        }
+        $res = $this->client->call(Client::METHOD_POST, '/account/sessions/email', [
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $id,
+        ], [
+            'email' => 'user@appwrite.io',
+            'password' => 'password'
+        ]);
+        $this->assertEquals('429', $res['headers']['status-code']);
+
+        $res = $this->client->call(Client::METHOD_POST, '/account/sessions/email', [
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $id,
+            'x-appwrite-development-key' => $devKey
+        ], [
+            'email' => 'user@appwrite.io',
+            'password' => 'password'
+        ]);
+        $this->assertEquals('401', $res['headers']['status-code']);
+
+
+        /**
+         * Test for FAILURE
```

Comment:
I would vote to add one more test that makes development key for 5 seconds, and after 5 second tries to use it.

It should either give exception, or hit rate limit after 10 attempts. Not sure which one is better tho, double check with Christy/Eldad what should be the behaviour

Reply:
1. This is creating a dev key that is already expired and making a request to make sure we get rate limit as we already mad 10 attempts above. what would another key that expires after 5 seconds do any different than this?

Reply:
Also added another test as you said, a key that expires after 5 seconds.

Reply:
I also added dev key to the existing client test so that we can keep the abuse enabled.